benchexec/tablegenerator/react-table/src/components/LinkOverlay.js

```diff
@@ -60,139 +60,156 @@ export default class LinkOverlay extends React.Component {
-    this.setState({ currentFile: url });
+  /*
+   * Loads the file of the given url. Four different approaches to load the file will be made in case the previous one fails:
+   * 1) AJAX request -> fails for ZIP archives
+   * 2) HTTP Range request for file in ZIP archive -> fails for ZIPs on the local disk
+   * 3) Normal HTTP request for file in ZIP archive -> fails for Google Chrome for ZIPs on the local disk
+   * 4) Manually via XMLHttpRequest
+   */
+  async loadFile(url) {
     if (url) {
+      this.setState({ currentFile: url });
       try {
         const response = await fetch(url);
-        //status 404/403 => error?
         if (isOkStatus(response.status)) {
-          try {
-            const content = await response.text();
-            this.setState({ content });
-          } catch (e) {
-            console.log("Error: Stream not readable", url, e);
-            this.attemptLoadingFromZIP(url);
-          }
+          const content = await response.text();
+          this.setState({ content });
         } else {
-          console.log("Error: Loading file not possible", response);
-          this.attemptLoadingFromZIP(url);
+          this.loadFileFromZip(url);
         }
       } catch (e) {
-        console.log("Error: Resource not found", url, e);
-        this.attemptLoadingFromZIP(url);
+        this.loadFileFromZip(url);
       }
     }
-  };
+  }
 
-  setError = (error) => {
-    this.setState({ error: `${error}` });
-  };
+  /* Loads the file from a ZIP archive and stores the entries in a cache for faster future access. */
+  loadFileFromZip(url) {
+    const decodedUrl = decodeURIComponent(url);
+    const folderSplitterSlash =
+      decodedUrl.lastIndexOf("/") > decodedUrl.lastIndexOf("\\") ? "/" : "\\";
+    const folderSplitPos = decodedUrl.lastIndexOf(folderSplitterSlash);
+    const zipPath = decodedUrl.substring(0, folderSplitPos) + ".zip";
+    const splittedUrl = decodedUrl.split(folderSplitterSlash);
+    const zipFile = `${splittedUrl[splittedUrl.length - 2]}/${
+      splittedUrl[splittedUrl.length - 1]
+    }`;
 
-  loadFileFromZipEntries = (entries, logfile, zipUrl) => {
-    for (let entry of entries) {
-      if (entry.filename.indexOf(logfile) >= 0) {
-        entry.getData(new zip.TextWriter(), (content) =>
-          this.setState({ content }),
-        );
-        return;
-      }
+    if (zipPath in zipEntriesCache) {
+      this.loadFileFromZipEntries(zipEntriesCache[zipPath], zipFile);
+    } else {
+      this.readZipArchiveHttpRange(zipPath, zipFile, true);
     }
-    this.setError(`Did not find file "${logfile}" in "${zipUrl}"`);
-  };
+  }
 
-  loadFileFromZip = (logZip, logfile, zipUrl) => {
-    logZip.getEntries((entries) => {
-      cachedZipFileEntries[zipUrl] = entries;
-      this.loadFileFromZipEntries(entries, logfile, zipUrl);
-    });
-  };
+  /* Tries to read the file from a ZIP archive with a HTTP range request.  */
+  readZipArchiveHttpRange(zipPath, zipFile) {
```

Comment:
I would rename this to `readZipArchive`. After all, it does not only use HTTP range requests, it also does the fallback.